### PR TITLE
Fix everything

### LINKED ISSUE #6: Copy to 0.4 network
@krl we need to put the data onto the 0.4 network, any pointers on how to easiest do this

cc @whyrusleeping @lgierth for ideas

### PR DESCRIPTION
**Pull request by dignifiedquire**
- Rewrite generator code
- Add source data to the repo (`data`)
- Add tests for generator code
- Refactor lookup code
- Add tests for lookup code
- Generate new data and add it to the 0.3 and 0.4 network (`QmRn43NNNBEibc6m7zVNcS6UusB1u3qTTfyoLmkugbeeGJ`)

Fixes #6

### CODE REVIEW

**Comment by dignifiedquire** [comment]
Travis fails because it's not pinned on the 0.3 network probably and the resolves are too slow due to that

File: README.md [hunk 1]
```diff
@@ -33,10 +38,10 @@ a `formatted` property that looks like this: `Mountain View, CA, United States,
 
 The utility geoip-gen reads csv files provided from GeoLite, and turns them into a 32-way branching b-tree, which is stored as ipfs json objects.
 
-There is a generator included, that can be called like this:
+There is a generator included, that can be run with
 
 ```bash
-$ node geoip-gen.js path/GeoLite-Blocks.csv path/GeoLite-Location.csv
```
Comment (daviddias) [inline README.md:39]: shall we add this script as a bin script to be installed and accessible globally? :)

**Comment by daviddias** [comment]
image
woooooo. Do we really want to push the dataset to the repo?

**Comment by daviddias** [comment]
Codewise LGTM :) great work @dignifiedquire !! Finally we have geoip working with proper tests and more docs :) :dancer:

**Comment by daviddias** [comment]
One request, if all that travis needs is to have the dataset pinned on 0.3, can we do that to see if it goes green? It will be also useful for anyone else that might want to contribute as well.

**Comment by dignifiedquire** [comment]
> woooooo. Do we really want to push the dataset to the repo?

I don't know, I'd rather have it checked in than go search for it again in the future tbh

File: README.md (hunk 1 above)
Comment (dignifiedquire) [inline README.md:39]: It's available as `npm run generate` as documented which I think is enough, but I'm okay adding it the bin scripts

**Comment by dignifiedquire** [comment]
> One request, if all that travis needs is to have the dataset pinned on 0.3, can we do that to see if it goes green? It will be also useful for anyone else that might want to contribute as well.

I want to make it green, but pinbot keeps dying when trying to pin it :(

**Comment by ghost** [comment]
> I don't know, I'd rather have it checked in than go search for it again in the future tbh

Let's put it into IPFS and pin.

```
This dataset: ipfs.io/ipfs/<hash1>
produces this merkledag: ipfs.io/ipfs/<hash2>
Latest dataset available here: someurl.com/dataset.tgz
```

**Comment by jbenet** [comment]
Yes, put it into IPFS, not this repo. We should pin it.

(We should look into making a pinbot that works with github repos.)

**Comment by dignifiedquire** [comment]
@lgierth @jbenet done, all data is now on ipfs, and gets downloaded automatically when generate is run :)

**Comment by ghost** [comment]
Great :+1: could you squash the commits so that we don't bloat the repo size unneccessarily?

**Comment by dignifiedquire** [comment]
@lgierth will do. any updates on pinning location data on 0.3? pinbot always dies

File: country_data/countries.csv [hunk 2]
```diff
@@ -1,260 +0,0 @@
-name,alpha2,countryCallingCodes,alpha3,ioc,currencies,languages,ccTLD,status
```
Comment (jbenet) [inline country_data/countries.csv:1]: where did this file go? is it not needed anymore?

**Comment by jbenet** [comment]
@dignifiedquire would be nice to make CI pass the tests, that way merging changes is way easier / safer. fine to be a new issue + separate PR.

File: country_data/countries.csv (hunk 2 above)
Comment (dignifiedquire) [inline country_data/countries.csv:1]: @jbenet onto ipfs

**Comment by dignifiedquire** [comment]
@jbenet they will pass when 0.3 has the geo data

**Comment by dignifiedquire** [comment]
Merging this for now, tests will work as soon as the 0.3 issues are resolved. But I need a new release for use in ipfs/webui.

**Comment by daviddias** [comment]
What is blocking us from pinning it on 0.3? Is there an issue in infrastructure to track that progress that can be linked from here?

**Comment by ghost** [comment]
> What is blocking us from pinning it on 0.3? Is there an issue in infrastructure to track that progress that can be linked from here?

nothing -- I'm on it

**Pull request merged**